pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BinaryOperatorTransformFunction.java

```diff
@@ -91,6 +105,91 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
     _rightTransformFunction = arguments.get(1);
     _leftStoredType = _leftTransformFunction.getResultMetadata().getDataType().getStoredType();
     _rightStoredType = _rightTransformFunction.getResultMetadata().getDataType().getStoredType();
+
+    if (_leftTransformFunction instanceof IdentifierTransformFunction
+        && _rightTransformFunction instanceof LiteralTransformFunction) {
+      IdentifierTransformFunction leftTransformFunction = (IdentifierTransformFunction) _leftTransformFunction;
+      if (leftTransformFunction.getDictionary() != null) {
+        LiteralTransformFunction rightTransformFunction = (LiteralTransformFunction) _rightTransformFunction;
+        if (rightTransformFunction.isNull()) {
+          _isNull = true;
+        }
+        String rightLiteralStr;
+        switch (_leftStoredType) {
+          case INT:
+            rightLiteralStr = Integer.toString(_isNull ? CommonConstants.NullValuePlaceHolder.INT
```

Comment:
You can use `LiteralTransformFunction.getIntLiteral()` to get the value. Same for other types